pkg/controllers/provisioning/dynamic/provisioner.go

```diff
@@ -135,29 +119,12 @@ func (p *Provisioner) Reconcile(ctx context.Context) (result reconcile.Result, e
 	if len(results.NewNodeClaims) == 0 {
 		return reconcile.Result{RequeueAfter: singleton.RequeueImmediately}, nil
 	}
-	if _, err = p.CreateNodeClaims(ctx, results.NewNodeClaims, WithReason(metrics.ProvisionedReason), RecordPodNomination); err != nil {
+	if _, err = p.ncProvisioner.CreateNodeClaims(ctx, results.NewNodeClaims, provisioning.WithReason(metrics.ProvisionedReason), provisioning.RecordPodNomination, provisioning.IgnoreStaticNodeClaims); err != nil {
```

Comment:
nit: I wonder if we should put static NodeClaims in some separate bucket in our results response since know we won't want to be creating them -- that would prevent us having to create this sort-of awkward option too

Reply:
This check is not needed, I can remove it. We anyway have a NC `IsStaticNode` check